logs.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixer
+
+import (
+	"encoding/json"
+	"io"
+	"time"
+)
+
+type (
+	// LogEntry is the top-level struct for logs data that will be
+	// exported by logging adapters.
+	LogEntry struct {
```

Comment:
I agree that the functionality may be useful outside of adapters, but that's kind of an edge case. I'd prefer to have it in the adapters directory so that an adapter author doesn't need to look beyond the adapters package to find what they need. Everything else can be considered "mixer guts" from their standpoint.